xapian-core/matcher/synonympostlist.h

```diff
@@ -58,6 +58,12 @@ class SynonymPostList : public PostList {
      */
     bool want_unique_terms;
 
+    /** Flag indicating whether the weighting object needs the max wdf of
+     *  terms.
+     *  FIXME : Not implemented
```

Comment:
A `SynonymPostList` has the same logical effect as a `OR` - i.e. it matches when any of its subqueries match.  But the weight it gives is as if it was a single term, and hence it's the `SynonymPostList` which gets the weight object, rather than the terms at the leaf of the query.

For example, you might want to treat `furze` and `gorse` as synonymous, so if the user searches for either term it will match documents indexed by either (or both), and a document which mentions `furze` 3 times would get the same weight as one which mentions `furze` twice, and `gorse` once, or `furze` one and `gorse` twice, etc (I think we may approximate slightly in some cases for efficiency, but the aim is that it works like this).

Calculating things like wdf, number of unique terms, or max_wdf for a `SynonymPostList` is potentially quite costly (since it requires fetching data from all its children) so we only do that for stats which are actually going to be used.

Looking at the code, it looks like `want_unique_terms` is currently always false here - looks like that was never hooked up properly either.  I think that just setting in `SynonymPostList::set_weight()`:

{{{
    want_unique_terms = wt->get_sumpart_needs_uniqueterms_();
}}}

That change compiles and passes the current testsuite, but we could do with a regression test to ensure that uniqueterms gets calculated for a `SynonymPostList` to ensure it's actually fixed by that change and that it stays fixed in the future.

Anyway, maxwdf should be handled similarly.